corehq/apps/domain/views/settings.py

```diff
@@ -550,54 +551,75 @@ def form(self):
         return DomainAlertForm()
 
     def post(self, request, *args, **kwargs):
-        command = request.POST.get('command')
-        if command:
-            self._apply_command(request, command)
-        elif self.form.is_valid():
+        if self.form.is_valid():
             self._create_alert()
             messages.success(request, _("Alert saved!"))
         else:
             messages.error(request, _("There was an error saving your alert. Please try again!"))
             return self.get(request, *args, **kwargs)
         return HttpResponseRedirect(self.page_url)
 
-    def _apply_command(self, request, command):
-        alert_id = request.POST.get('alert_id')
-        assert alert_id, 'Missing alert ID'
-        alert = self._load_alert(alert_id)
-        if not alert:
-            messages.error(request, _("Alert not found!"))
-            return
-
-        if command == 'delete':
-            alert.delete()
-        elif command in ['activate', 'deactivate']:
-            self._update_alert(alert, command)
-            messages.success(request, _("Alert updated!"))
-        else:
-            messages.error(request, _("Unexpected update received. Alert not updated!"))
-
-    @staticmethod
-    def _update_alert(alert, command):
-        if command == 'activate':
-            alert.active = True
-        elif command == 'deactivate':
-            alert.active = False
-        alert.save(update_fields=['active'])
-
-    def _load_alert(self, alert_id):
-        try:
-            return Alert.objects.get(
-                created_by_domain=self.domain,
-                id=alert_id
-            )
-        except Alert.DoesNotExist:
-            return None
-
     def _create_alert(self):
         Alert.objects.create(
             created_by_domain=self.domain,
             domains=[self.domain],
             text=self.form.cleaned_data['text'],
             created_by_user=self.request.couch_user.username,
         )
+
+
+@toggles.CUSTOM_DOMAIN_BANNER_ALERTS.required_decorator()
+@domain_admin_required
+@require_POST
+def update_domain_alert_status(request, domain):
+    alert_id = request.POST.get('alert_id')
+    assert alert_id, 'Missing alert ID'
+    alert = _load_alert(alert_id, domain)
+    if not alert:
+        messages.error(request, _("Alert not found!"))
+    else:
+        command = request.POST.get('command')
+        if command:
+            _apply_command(request, alert, command)
```

Comment:
The result would appear to be more consistent without the conditional
```suggestion
        command = request.POST.get('command')
        _apply_command(request, alert, command)
```